File: app_mobile/kmp/androidApp/src/main/java/com/ansmi/gestsquadre/kmp/map/WaypointIcons.kt
package com.ansmi.gestsquadre.kmp.map

import androidx.annotation.DrawableRes
import com.ansmi.gestsquadre.kmp.R

object WaypointIcons {
    @DrawableRes
    fun drawableRes(iconKey: String?): Int =
        when (iconKey?.trim().orEmpty()) {
            "croce_rossa" -> R.drawable.waypoint_croce_rossa
            "club_house" -> R.drawable.waypoint_club_house
            "cancello_in" -> R.drawable.waypoint_cancello_in
            "driving_range" -> R.drawable.waypoint_driving_range
            "villaggio_comm" -> R.drawable.waypoint_villaggio_comm
            "welcome" -> R.drawable.waypoint_welcome
            "media_center" -> R.drawable.waypoint_media_center
            "toc" -> R.drawable.waypoint_toc
            "buche", "buca_golf", "" -> R.drawable.waypoint_buche
            else -> R.drawable.waypoint_buche
        }
}

File: app_mobile/kmp/iosApp/iosApp/WaypointIcons.swift
import Foundation

enum WaypointIcons {
    static func imageName(for iconKey: String?) -> String {
        switch iconKey?.trimmingCharacters(in: .whitespacesAndNewlines) ?? "" {
        case "croce_rossa": return "waypoint_croce_rossa"
        case "club_house": return "waypoint_club_house"
        case "cancello_in": return "waypoint_cancello_in"
        case "driving_range": return "waypoint_driving_range"
        case "villaggio_comm": return "waypoint_villaggio_comm"
        case "welcome": return "waypoint_welcome"
        case "media_center": return "waypoint_media_center"
        case "toc": return "waypoint_toc"
        case "pma": return "waypoint_pma"
        case "buche", "buca_golf", "": return "waypoint_buche"
        default: return "waypoint_buche"
        }
    }
}
